### DISCUSSION TITLE
Problems upgrading 1.4.5 to 1.5.0

### DISCUSSION THREAD

**Discussion by dpasman** [siderolabs/talos#7710] [Q&A]
Hi all,

I am having problems upgrading talos from 1.4.5 to 1.5.0. I have tried upgrading multiple times with a testcluster I am creating via Terraform on Proxmox. The issue occurs on a 3 node controlplane cluster.

I am running the following command for every server one by one:
talosctl upgrade --nodes <node_ip> --image ghcr.io/siderolabs/installer:v1.5.0

The first 2 nodes succeeded the upgrade. I am running talosctl health after every node to make sure everything is up and running. But when upgrading the last node of the 3. I get the following error:

```
◱ watching nodes: [10.100.20.26]
    * 10.100.20.26: 1 error(s) occurred:
    sequence error: sequence failed: error running phase 4 in upgrade sequence: task 1/1: failed, failed to leave cluster: 1 error(s) occurred:
    failed to remove member 14306642156463998993: etcdserver: server stopped
```

So I tried doing the same multiple times, but everytime it seem to fail on the error above.

Is this a known issue? Are there any other people having this issue perhaps?

Thank you in advance!

**Comment by smira** [comment]
There is a very similar issue #7423, but I can't find the root cause of it.

If you have a reproducer, can you please help me to debug this further?

What I'm looking for is something like the following:

1. `talosctl health`  (before an upgrade to make sure all good)
2. `talosctl etcd members` (to see member IDs before an upgrade)
3. `talosctl logs -f -n <cp1>,<cp2>,... etcd` (keep this running so that you can grab etcd logs during the upgrade)
4. `talosctl dmesg -f -n <node-to-be-upgraded>` (keep this running so you have the logs up to the moment upgrade fails)
5. kick off an upgrade... and please attach the logs to the ticket

**Reply by smira** [reply]
And one more thing: `talosctl -n <cp1>,<cp2>,<cp3> get etcdmember`

**Comment by dpasman** [comment]
Hereby the output of the commands:

talosctl health -n 10.100.20.28:
```discovered nodes: ["10.10.42.6" "10.10.42.4" "10.10.42.5"]
waiting for etcd to be healthy: ...
waiting for etcd to be healthy: OK
waiting for etcd members to be consistent across nodes: ...
waiting for etcd members to be consistent across nodes: OK
waiting for etcd members to be control plane nodes: ...
waiting for etcd members to be control plane nodes: OK
waiting for apid to be ready: ...
waiting for apid to be ready: OK
waiting for all nodes memory sizes: ...
waiting for all nodes memory sizes: OK
waiting for all nodes disk sizes: ...
waiting for all nodes disk sizes: OK
waiting for kubelet to be healthy: ...
waiting for kubelet to be healthy: OK
waiting for all nodes to finish boot sequence: ...
waiting for all nodes to finish boot sequence: nodes ["10.10.42.6"] are still in boot sequence
waiting for all nodes to finish boot sequence: OK
waiting for all k8s nodes to report: ...
waiting for all k8s nodes to report: OK
waiting for all k8s nodes to report ready: ...
waiting for all k8s nodes to report ready: OK
waiting for all control plane static pods to be running: ...
waiting for all control plane static pods to be running: OK
waiting for all control plane components to be ready: ...
waiting for all control plane components to be ready: OK
waiting for kube-proxy to report ready: ...
waiting for kube-proxy to report ready: OK
waiting for coredns to report ready: ...
waiting for coredns to report ready: OK
waiting for all k8s nodes to report schedulable: ...
waiting for all k8s nodes to report schedulable: OK 
```

I want to note that the node IPs I connect with are in another range, the range given in "discovered nodes" is another network, used for storage. I am not sure if this may give some issues while upgrading.

talosctl etcd members -n 10.100.20.28:
```
NODE           ID                 HOSTNAME               PEER URLS                 CLIENT URLS               LEARNER
10.100.20.28   322527cb025f7535   test2-controlplane-1   https://10.10.42.6:2380   https://10.10.42.6:2379   false
10.100.20.28   3bfcea9ec9743370   test2-controlplane-2   https://10.10.42.4:2380   https://10.10.42.4:2379   false
10.100.20.28   ea7656a15f8173ea   test2-controlplane-3   https://10.10.42.5:2380   https://10.10.42.5:2379   false
```

❯ talosctl -n 10.100.20.28,10.100.20.26,10.100.20.25 get etcdmember
```
NODE           NAMESPACE   TYPE         ID      VERSION   MEMBER ID
10.100.20.28   etcd        EtcdMember   local   1         b7b47915089d6b76
10.100.20.26   etcd        EtcdMember   local   1         3bfcea9ec9743370
10.100.20.25   etcd        EtcdMember   local   1         ea7656a15f8173ea
```

Tried executing the whole upgrade now and it worked, I will try it again but wanted you let you know the fact we may have some issues with the client urls being the storage network.

**Reply by smira** [reply]
If you have multiple networks, make sure you have configuration set properly: https://www.talos.dev/v1.5/reference/configuration/#etcdconfig

Also Kubernetes pod/service CIDRs should match whatever is used in the cluster.

**Reply by smira** [reply]
Also I see a problem with member IDs...

```
10.100.20.28   etcd        EtcdMember   local   1         b7b47915089d6b76
```

```
10.100.20.28   322527cb025f7535   test2-controlplane-1   https://10.10.42.6:2380   https://10.10.42.6:2379   false
```

**Reply by smira** [reply]
So if etcd is using some ephemeral addresses (e.g. ones which might be removed on node shutdown), this might lead to very surprising result.

**Reply by dpasman** [reply]
Thanks for your replies. I will look into this and try to solve this first before troubleshooting any further. It will probably take me around 2 weeks because of my holidays 😄 

Does the etcd advertisesubnet needs to be the same as the Node frontend network? (The network I am connecting from to talk to the TalosAPI?) etcd did not use some empheral address, it is just an address outside of our frontend-network.

**Reply by smira** [reply]
Advertise address is the address etcd advertises to the peers (other etcd members) to connect to. Listen address is what etcd listens on, it should contain adverised address of course, Talos will automatically add localhost there as well. 

So you usually listen and advertise private subnet (if any), as you don't want etcd to be even listening on public IPs.
